packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -318,14 +318,64 @@ class DateTimePicker extends React.Component {
 
   handleTimeChange(event, time) {
     this.timeValue = time;
-    const validDate = DateUtil.isValidDate(this.dateValue, this.state.dateFormat) && this.isDateTimeAcceptable(DateTimeUtils.convertDateTimeStringToMomentObject(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds, this.initialTimeZone));
+    let validDate = DateUtil.isValidDate(this.dateValue, this.state.dateFormat) && this.isDateTimeAcceptable(DateTimeUtils.convertDateTimeStringToMomentObject(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds, this.initialTimeZone));
     const validTime = DateTimeUtils.isValidTime(this.timeValue, this.props.showSeconds);
-    const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+    let previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+
+    if (event.type === 'keydown') {
+      // Keyboard shortcut is pressed, split the time value into hour and minute inputs
+      const timeArray = this.timeValue.split(':');
+      if (event.key === 'n' || event.key === 'N') {
+        // `N` set Date to today
+        previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+        validDate = true;
+      } else if (event.key === '-' || event.key === '_') {
+        // `-` subtract 1 minute`
+        if (!validDate) {
+          // Date is not valid, set date to today making it valid
+          previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+          validDate = true;
+        }
+
+        if (timeArray[0] === '23' && timeArray[1] === '59') {
+          // Time is 23:59 which means previous datetime should be on the wrong day, lower it by 1
+          previousDateTime = previousDateTime.subtract(1, 'd');
+        }
+
+        const changedDateTime = DateTimeUtils.updateTime(previousDateTime, this.timeValue, this.props.showSeconds);
+        const checkHour = previousDateTime.subtract(1, 'minute').hours();
+        previousDateTime.add(1, 'minute');
```

Comment:
Because the `add` and `subtract` APIs mutates the original moment and we have no intention to change the value of `previousDateTime` here, instead of subtracting a minute and then adding a minute back, it seems to make more sense and cleaner to just make a clone of `previousDateTime` to use for `checkHour`.

Something like this to prevent having to change the value of previousDateTime.

> const checkHour = previousDateTime.**clone()**.subtract(1, 'minute').hours();

Reply:
Changed here https://github.com/cerner/terra-framework/pull/1520/commits/d798c38e2975d58fd2abecc55bc87db4aed4ac68#diff-063daa4349be2bbad9c58209d280bfaee7a04b7e429163ebdb7784768f501a5cR358